pantry/pantry_method_script.sh

```diff
@@ -0,0 +1,19 @@
+#!/bin/bash
+
+set -o errexit
+set -o pipefail
+
+DATALINK="$(svcprop -c -p config/datalink "${SMF_FMRI}")"
+GATEWAY="$(svcprop -c -p config/gateway "${SMF_FMRI}")"
```

Comment:
I think we should probably explicitly check that the `DATALINK` and `GATEWAY` values here are not set to the `unknown` default value?

After L3-4, but before L6, we could insert:

```sh
. /lib/svc/share/smf_include.sh
```

Then, if we detect missing configuration, we can exit `$SMF_EXIT_ERR_CONFIG` to have SMF flag a permanent configuration error; e.g.,

```sh
if [[ $DATALINK == unknown ]] || [[ $GATEWAY == unknown ]]; then
        printf 'ERROR: missing datalink or gateway\n' >&2
        exit "$SMF_EXIT_ERR_CONFIG"
fi
```

Reply:
Sure, updated.

Couple things:
- Is there a standard for checking "empty values" of properties in illumos? I noticed that I could either...
  - (1) totally delete the property, which obfuscates the value by removing it from the manifest.xml file
  - (2) set it to `''`, but then it appears to be set in the shell to `\"\"`, rather than an actual empty string, which is kinda odd.
  - (3) use this quirky naming scheme, at least for `astring` values, of setting it to some reserved sentinel value (like `default`).

I ask because if we generalize some of this stuff to an extra service, I don't *think* all these values actually need to be set, so some could be only supplied optionally. For example, I think the underlay address we're supplying might not always be available when the zone starts (such as within the switch zone).